sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/aio/_client.py

```diff
@@ -61,20 +62,23 @@ def _get_attributes(self, enabled, not_before, expires_on, exportable=None):
             )
         return None
 
-    def get_cryptography_client(self, key):
-        # type: (str) -> CryptographyClient
+    def get_cryptography_client(self, key_name, **kwargs):
+        # type: (str, **Any) -> CryptographyClient
         """Gets a :class:`~azure.keyvault.keys.crypto.aio.CryptographyClient` for the given key.
 
-        :param str key: The full identifier of the key used to perform cryptographic operations. This must include the
-            key version.
+        :param str key_name: The name of the key used to perform cryptographic operations.
+
+        :keyword str version: Optional version of the key used to perform cryptographic operations.
 
         :returns: A :class:`~azure.keyvault.keys.crypto.aio.CryptographyClient` using the same options, credentials, and
             HTTP client as this :class:`~azure.keyvault.keys.aio.KeyClient`.
         :rtype: ~azure.keyvault.keys.crypto.aio.CryptographyClient
         """
+        key_id = _get_key_id(self._vault_url, key_name, kwargs.get("version"))
+
         # We provide a fake credential because the generated client already has the KeyClient's real credential
         return CryptographyClient(
-            key, object(), generated_client=self._client, generated_models=self._models  # type: ignore
+            key_id, object(), generated_client=self._client, generated_models=self._models  # type: ignore
```

Comment:
Maybe I missed it, but just to confirm - this `generated_client` will be the one that is used to create a cryptoClient? And do we skip instantiating a throwaway `generated_client` in that case?

Reply:
Yep! We just directly set the generated client of the resulting CryptographyClient to be the one we pass in here. The tests for `get_cryptography_client` check that the `._client` of both clients reference the same object (using an `==` equality, rather than just checking for properties)